### ISSUE TITLE
[css-forms-1] WCAG conformance design principle

### ISSUE THREAD

**Issue by lukewarlow** [w3c/csswg-drafts#12480]
This issue is to discuss potential clarifications to the "The controls pass 100% of [WCAG 2.2 AA standards](https://www.w3.org/TR/WCAG22/)." design principle of CSS Forms base appearance mode.

This is strongly worded and while a principle rather than a claim I think it could be good to clarify the scope of what is and isn't included in that. For example, the title attribute tooltip that shows up has a number of WCAG issues (that it avoids by virtue of the fact its browser provided).

Pickers are also largely not included in the spec but the statement might give a false sense that this sovles all those issues.

Potentially we should simply say something more generic that the controls should try to account for accessibility requirements where possible.

Raising as its something that's come up on socials (e.g. https://toot.cafe/@aardrian/114851791346613604).

**Comment by lukewarlow** [comment]
Personally, I think we should drop the 100% number as some will be contextual and require authors do extra stuff in certain circumstances regardless. Though I equally think we should definitely have a principle related to accessibility. Even if it's that we can't rely on being "UA provided" as a scapegoat for failing requirements.

We also won't be able to prevent authors from breaking these accessibility affordances, so that's another clarification that could be useful.

**Comment by nt1m** [comment]
@aardrian These design principles were specifically about `appearance: base` default styles, nothing else. I don't know if that was clear from the spec, but if not, we should clarify that.

I agree "100%" isn't doing service here, especially since the design principles are sometimes applied in different orders of importance (with a different principle sometimes contradicting with the accessibility one).

I believe the original aim of this principle when we originally thought it out at Apple was to say that developers should be able to use a form control with `appearance: base` (and nothing else) without worrying about accessibility because accessibility comes out of the box. But yeah, it's probably idealistic and not a good message to give developers to say that they don't need to think about it.

The principles in general were meant to be strongly worded (e.g. no conditional), to make sure we address underlying issues are addressed.

Perhaps it's better to word it differently:

> The controls are accessible out of the box in default configurations, aiming to pass WCAG 2.2 AA standards.

and perhaps add a normative note about accessibility:

> NOTE: Authors may need to do extra adjustments to ensure their controls are accessible depending on the context the form control is placed into. The basic appearance does not prevent adjustments by the author that are inaccessible.

 (or something among those lines)

cc @fantasai

**Comment by aardrian** [comment]
@nt1m 
> These design principles were specifically about appearance: base default styles, nothing else. I don't know if that was clear from the spec, but if not, we should clarify that.

It was clear to me from the conversation on the socials.

> But yeah, it's probably idealistic and not a good message to give developers to say that they don't need to think about it.

+1. And I know that wasn't the intent, but people will tend to read what they want into a thing. Heck, even the social post that kicked this discussion of lead by asserting this spec _will_ make things auto-conform. So someone somewhere has already mis-read it.

You might want to consider tweaking the principal itself from "The controls pass..." to something like "The controls target..." Your suggestion is nearly that (I only changed one word and am not even sure it's a good word).

I mean, I _very_ much want them to come out of that gate passing at 100% (and, frankly, think if they can't then this is probably wasted effort), but think a qualifier puts you in position to under-promise and over-deliver.

**Comment by fantasai** [comment]
> The controls are accessible out of the box in default configurations, aiming to pass WCAG 2.2 AA standards.

I think this captures it, maybe s/aiming to pass/ideally passing/

> and perhaps add a normative note about accessibility:

Notes are always informative, by definition. :)

> NOTE: Authors may need to do extra adjustments to ensure their controls are accessible depending on the context the form control is placed into. The basic appearance does not prevent adjustments by the author that are inaccessible.

s/the context/any customizations as well as the context/ ?

**Comment by lukewarlow** [comment]
I've opened a PR to make this change. Sorry for the delay.
